### PR TITLE
Fix two leaks in X509_REQ_to_X509

### PR DESCRIPTION
**Pull request by FdaSilvaYY**
Fix of Issue #182.
I wish, I had a X509_REQ_get0_pubkey method, so fix could be shorter.

### CODE REVIEW

**Comment by kaduk** [comment]
> I wish, I had a X509_REQ_get0_pubkey method, so fix could be shorter.

So make one!  (In a separate commit.)

File: crypto/x509/x509_req.c [hunk 1]
```diff
@@ -115,6 +115,13 @@ EVP_PKEY *X509_REQ_get_pubkey(X509_REQ *req)
     return (X509_PUBKEY_get(req->req_info.pubkey));
 }
 
+EVP_PKEY *X509_REQ_get0_pubkey(X509_REQ *req)
+{
+    if (req == NULL)
+        return (NULL);
```
Comment (richsalz) [inline crypto/x509/x509_req.c]: remove the parens.  i prefer using ```?:``` but I think most don't so that part is okay.

**Comment by richsalz** [comment]
+1.  @mattcaswell you have a view?

File: apps/req.c [hunk 2]
```diff
@@ -375,6 +375,7 @@ int req_main(int argc, char **argv)
     if (!nmflag_set)
         nmflag = XN_FLAG_ONELINE;
 
+    // pkey is always NULL here, no ?
```
Comment (richsalz) [inline apps/req.c]: C++ comment.

Comment (FdaSilvaYY) [inline apps/req.c]: Ok, ficed.
It's just a side note, I don't dig this code line origins.

**Comment by richsalz** [comment]
done, thanks!

**Pull request closed without merging**